### PR TITLE
allow hyphens in uniprot protein accessions

### PR DESCRIPTION
**Pull request by rmillikin**

### CODE REVIEW

File: UsefulProteomicsDatabases/ProteinDbLoader.cs [hunk 1]
```diff
@@ -14,7 +14,7 @@ namespace UsefulProteomicsDatabases
 {
     public static class ProteinDbLoader
     {
-        public static readonly FastaHeaderFieldRegex UniprotAccessionRegex = new FastaHeaderFieldRegex("accession", @"([A-Z0-9_.]+)", 0, 1);
+        public static readonly FastaHeaderFieldRegex UniprotAccessionRegex = new FastaHeaderFieldRegex("accession", @"([A-Z0-9_.-]+)", 0, 1);
         public static readonly FastaHeaderFieldRegex UniprotFullNameRegex = new FastaHeaderFieldRegex("fullName", @"\s(.*?)\sOS=", 0, 1);
         public static readonly FastaHeaderFieldRegex UniprotNameRegex = new FastaHeaderFieldRegex("name", @"\|([^\|][A-Z0-9_]+)", 1, 1);
         public static readonly FastaHeaderFieldRegex UniprotGeneNameRegex = new FastaHeaderFieldRegex("geneName", @"GN=([^ ]+)", 0, 1);
```
Comment (rmmiller22) [inline UsefulProteomicsDatabases/ProteinDbLoader.cs:20]: can you also add the dash here
public static readonly FastaHeaderFieldRegex EnsemblAccessionRegex = new FastaHeaderFieldRegex("accession", @"([A-Z0-9_.-]+)", 0, 1);

Comment (rmillikin) [reply UsefulProteomicsDatabases/ProteinDbLoader.cs:20]: yes but metamorpheus will not use it. MM uses uniprot regex for everything without the ability to change it. https://github.com/smith-chem-wisc/MetaMorpheus/blob/a8f1201d0ff83040dbc25ae2625ae7f5ed996bb9/TaskLayer/MetaMorpheusTask.cs#L540

Comment (rmmiller22) [reply UsefulProteomicsDatabases/ProteinDbLoader.cs:20]: Okay got it. No worries then. Fine as is

**Review by rmmiller22 — APPROVED** [review]

**Comment by rmmiller22** [comment]
Appveyor is failing but code looks good as far as I can tell

**Comment by trishorts** [comment]
i tested this and did not get any unit test failures. looks good to me.

**Review by trishorts — APPROVED** [review]
i'm good if you can get appveyor to stop failing.

**Review by trishorts — APPROVED** [review]
deleted test/bin/debug.
checkout PR
image
all unit tests passed

**Review by rmmiller22 — APPROVED** [review]

**Pull request merged**